app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/templates.tsx

```diff
@@ -0,0 +1,144 @@
+    },
+]
+
+/** Render all templates, includes empty template */
+interface TemplatesRenderProps {
+    // Later this data may be requested and therefore needs to be passed dynamically.
+    templates: Template[]
+    onTemplateClick(name?: string): void
+}
+
+const TemplatesRender: React.FC<TemplatesRenderProps> = ({ templates, onTemplateClick }) => {
+    /** Unify the border color and the text color. */
+
+    const borderColor = '9E8C91'
+    /**
+     * Dash border spacing is not supported by native CSS.
+     * Therefore, use a background image to create the border.
+     * It is essentially an SVG image that was generated by the website.
+     * @see {@link https://kovart.github.io/dashed-border-generator}
+     */
+    const borderBgImg = `url("data:image/svg+xml,%3csvg width='100%25' height='100%25' xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' rx='16' ry='16' stroke='%23${borderColor}FF' stroke-width='4' stroke-dasharray='15%2c 15' stroke-dashoffset='0' stroke-linecap='butt'/%3e%3c/svg%3e")`
+
+    /** The action button for creating an empty project.
+     * So there is no need to pass a value to onChange function.
+     */
+    const CreateEmptyTemplate = (
+        <button
+            className="h-40 cursor-pointer"
+            onClick={() => {
+                onTemplateClick()
+            }}
+        >
+            <div
+                style={{ backgroundImage: borderBgImg }}
+                className={`flex h-full w-full rounded-2xl text-[#${borderColor}]`}
+            >
+                <div className="m-auto text-center">
+                    <button>
+                        <svg
+                            xmlns="http://www.w3.org/2000/svg"
+                            fill="none"
+                            viewBox="0 0 24 24"
+                            strokeWidth={0.5}
+                            stroke="currentColor"
+                            className="w-20 h-20"
+                        >
+                            <path
+                                strokeLinecap="round"
+                                strokeLinejoin="round"
+                                d="M12 9v6m3-3H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
+                            />
+                        </svg>
+                    </button>
+                    <p className="font-semibold text-sm">New empty project</p>
+                </div>
+            </div>
+        </button>
+    )
+
+    return (
+        <>
+            {CreateEmptyTemplate}
+            {templates.map(template => (
+                <button
+                    key={template.title}
+                    className="h-40 cursor-pointer"
+                    onClick={() => {
+                        onTemplateClick(template.templateName)
+                    }}
+                >
+                    <div
+                        // style={{ backgroundImage }}
```

Comment:
commented code should either be removed, or a `TODO` or `FIXME` comment (whichever is more appropriate) should be added explaining why it's commented and how it will be fixed later